src/components/templates/Audience/Audience.tsx

```diff
@@ -198,6 +198,8 @@ export const Audience: React.FunctionComponent = () => {
 
   // These are going to be translated (ie. into negative/positive per above)
   // That way, when the audience size is expanded these people keep their seats
+
+  // FIXME: This is really bad, needs to be fixed ASAP
```

Comment:
A potential, larger fix, for the entire Audience template would be to not use `<div className="seat-row">` for each row, that is populated with `<div className="seat">` but to use a true css grid.

The user would then be assigned a class `seat-2-1` for instance, to denote that the user is on the *second row* *column 1* or vice-versa, which would mean he/she/it is sitting in the `grid-area: ....... "seat-2-1"`

Potential downfall for this is that the `grid-area` would be generated every time the admin changes row/column amount in the panel which would require a *refresh* on the users part

Reply:
The way I had been intending to solve this (though never was able to get the spare time to do so) was to re-implement it using `canvas` instead of the 'DOM grid'.

This was briefly mentioned in https://trello.com/c/T70bzhZF and then agreed as the probably 'best direction' in later discussions (eg. [slack thread](https://doesitsparkle.slack.com/archives/C01ER4M61TL/p1606126714222700))/tickets such as https://trello.com/c/YqwUY91W

Reply:
I was referring to the way `const partygoersBySeat` was populated

`  const partygoersBySeat: WithId<User>[][] = [];
  let seatedPartygoers = 0;
  partygoers?.forEach((partygoer) => {
    if (
      !venueId ||
      !partygoer?.data ||
      partygoer.data[venueId] === undefined ||
      partygoer.data[venueId].row === undefined ||
      partygoer.data[venueId].column === undefined
    )
      return;
    const row = partygoer.data[venueId].row || 0;
    const column = partygoer.data[venueId].column || 0;
    if (!(row in partygoersBySeat)) {
      partygoersBySeat[row] = [];
    }
    partygoersBySeat[row][column] = partygoer;
    seatedPartygoers++;
  });`

If we were to refactor it, this logic should have been extracted into a separate function and wrapper into a memo hook

Reply:
If we were to refactor it, I would move it to a function and wrapped it into a memo hook

![image](https://user-images.githubusercontent.com/63194656/103632384-e3a0b100-4f4c-11eb-89bd-d90dc3296b82.png)

Reply:
Thinking about this a little more, I believe I already refactored this code in other places into `usePartygoersBySeat`:

- https://github.com/sparkletown/sparkle/blob/staging/src/components/templates/Camp/hooks/usePartygoersBySeat.ts`

Haven't deeply compared, but pretty sure it's the same thing.

Reply:
I can't access the link, could it be broken?

Reply:
It had an extra ` on the end, just edited it to fix it up.